ably/options.go

```diff
@@ -1070,6 +1074,14 @@ func WithLogLevel(level LogLevel) ClientOption {
 	}
 }
 
+// WithAgents is used to add product/version key-value pairs to include in the
+// agent library identifiers. This must only be used by Ably-authored SDKs.
+func WithAgents(agents map[string]string) ClientOption {
+	return func(os *clientOptions) {
+		os.Agents = agents
```

Comment:
Instead of passing by reference, can we do `copy by value`, create a new object for agents and use it instead.

Reply:
https://david-yappeter.medium.com/golang-pass-by-value-vs-pass-by-reference-e48aac8b2716

Reply:
I was just keeping to the same style as the other params eg `WithTransportParams` just passes the map by reference

Reply:
I think it's better to keep the same style